packages/core/src/browser/shell/tab-bar-decorator.ts

```diff
@@ -0,0 +1,175 @@
+    protected readonly contributions: ContributionProvider<TabBarDecorator>;
+
+    protected decorators: ReadonlyArray<TabBarDecorator>;
+
+    @postConstruct()
+    protected init(): void {
+        this.decorators = this.contributions.getContributions();
+        this.toDispose.pushAll(this.decorators.map(decorator =>
+            decorator.onDidChangeDecorations(data =>
+                this.onDidChangeDecorationsEmitter.fire(undefined)
+            ))
+        );
+
+    }
+
+    dispose(): void {
+        this.toDispose.dispose();
+    }
+
+    getDecorations(titles: Title<Widget>[]): Map<string, TabBarDecoration.Data[]> {
+        const changes: Map<string, TabBarDecoration.Data[]> = new Map();
+        for (const decorator of this.decorators) {
+            console.log(decorator);
+            for (const [id, data] of (decorator.decorate(titles)).entries()) {
+                if (changes.has(id)) {
+                    changes.get(id)!.push(data);
+                } else {
+                    changes.set(id, [data]);
+                }
+            }
+        }
+        return changes;
+    }
+}
+
+export namespace TabBarDecoration {
+
+    export namespace Styles {
+        export const CAPTION_HIGHLIGHT_CLASS = 'theia-caption-highlight';
+        export const CAPTION_PREFIX_CLASS = 'theia-caption-prefix';
+        export const CAPTION_SUFFIX_CLASS = 'theia-caption-suffix';
+        export const ICON_WRAPPER_CLASS = 'theia-icon-wrapper';
+        export const DECORATOR_SIZE_CLASS = 'theia-decorator-size';
+        export const TOP_RIGHT_CLASS = 'theia-top-right';
+        export const BOTTOM_RIGHT_CLASS = 'theia-bottom-right';
+        export const BOTTOM_LEFT_CLASS = 'theia-bottom-left';
+        export const TOP_LEFT_CLASS = 'theia-top-left';
+    }
+
+    /**
+     * Enumeration for the quadrant to overlay the icon on.
+     */
+    export enum IconOverlayPosition {
+        TOP_RIGHT,
+        BOTTOM_RIGHT,
+        BOTTOM_LEFT,
+        TOP_LEFT
+    }
+
+    export type Color = string;
+
+    export namespace IconOverlayPosition {
+
+        export function getStyle(position: IconOverlayPosition): string {
+            switch (position) {
+                case IconOverlayPosition.TOP_RIGHT: return TabBarDecoration.Styles.TOP_RIGHT_CLASS;
+                case IconOverlayPosition.BOTTOM_RIGHT: return TabBarDecoration.Styles.BOTTOM_RIGHT_CLASS;
+                case IconOverlayPosition.BOTTOM_LEFT: return TabBarDecoration.Styles.BOTTOM_LEFT_CLASS;
+                case IconOverlayPosition.TOP_LEFT: return TabBarDecoration.Styles.TOP_LEFT_CLASS;
+            }
+        }
+    }
+
+    /**
+     * A shape that can be optionally rendered behind the overlay icon. Can be used to further refine colors.
+     */
+    export interface IconOverlayBackground {
+
+        readonly shape: 'circle' | 'square';
+        readonly color?: Color;
+    }
+
+    /**
+     * Has no effect if the tree node being decorated has no associated icon.
```

Comment:
It looks like the documentation may have been copied over  from the `TreeDecorator`.
Please make sure to adjust the documentation in the PR.